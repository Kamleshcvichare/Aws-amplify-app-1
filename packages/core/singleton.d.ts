export * from './lib-esm/Singleton';