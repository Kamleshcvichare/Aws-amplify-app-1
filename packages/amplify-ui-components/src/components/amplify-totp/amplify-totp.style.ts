import { css } from 'emotion';

export const totp = css`
  text-align: center;
  margin-bottom: 30px;

  img {
    height: 200px;
  }
`;
