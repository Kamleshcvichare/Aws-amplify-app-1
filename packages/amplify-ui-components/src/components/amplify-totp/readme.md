# amplify-totp

<!-- Auto Generated Below -->


## Properties

| Property   | Attribute | Description                                                            | Type                   | Default     |
| ---------- | --------- | ---------------------------------------------------------------------- | ---------------------- | ----------- |
| `MFATypes` | --        | Types of MFA options                                                   | `MFATOTPOptions`       | `undefined` |
| `authData` | --        | Current authenticated user in order to sign requests properly for TOTP | `CognitoUserInterface` | `null`      |


## Dependencies

### Used by

 - [amplify-examples](../amplify-examples)

### Depends on

- [amplify-form-section](../amplify-form-section)
- [amplify-radio-button](../amplify-radio-button)

### Graph
```mermaid
graph TD;
  amplify-totp --> amplify-form-section
  amplify-totp --> amplify-radio-button
  amplify-form-section --> amplify-button
  amplify-form-section --> amplify-section
  amplify-radio-button --> amplify-label
  amplify-examples --> amplify-totp
  style amplify-totp fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
