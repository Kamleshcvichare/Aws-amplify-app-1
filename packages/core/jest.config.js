module.exports = {
	...require('../../jest.config'),
	coverageThreshold: {
		global: {
			branches: 74, // todo(ashwinkumar6): temp, add tests and bump back
			functions: 72,
			lines: 87, // todo(ashwinkumar6): temp, add tests and bump back
			statements: 88,
		},
	},
};
