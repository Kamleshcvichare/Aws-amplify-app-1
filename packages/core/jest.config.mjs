import { getJestConfig } from '../../jest/getJestConfig.mjs';
import tsconfig from './tsconfig.json' assert { type: 'json' };

const jestConfig = getJestConfig(tsconfig.compilerOptions, {
	coverageThreshold: {
		global: {
			branches: 75,
			functions: 72,
			lines: 88,
			statements: 88,
		},
	},
});

export default jestConfig;
