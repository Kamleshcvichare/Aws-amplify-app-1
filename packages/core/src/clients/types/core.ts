/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
