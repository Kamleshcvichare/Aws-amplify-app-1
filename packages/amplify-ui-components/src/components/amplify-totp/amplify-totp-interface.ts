import { CognitoUserInterface } from '../../common/types/auth-types';

export interface TOTPInterface {
  authData?: CognitoUserInterface;
  onTOTPEvent?: (event: any, data: any, user: any) => void;
  theme?: any;
}

export interface TOTPInterfaceState {
  code: string | null;
  setupMessage: string | null;
}
