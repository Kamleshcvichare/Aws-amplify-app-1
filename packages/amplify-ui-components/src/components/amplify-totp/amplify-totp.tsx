import { Component, Prop, State, h } from '@stencil/core';
import { Logger } from '@aws-amplify/core';
import { CognitoUserInterface, MFATOTPOptions } from '../../common/types/auth-types';
// import { Auth } from '@aws-amplify/auth';

const logger = new Logger('TOTP');

@Component({
  tag: 'amplify-totp',
  shadow: false,
})
export class AmplifyTOTP {
  @Prop() authData: CognitoUserInterface = null;
  @Prop() MFATypes: MFATOTPOptions;

  @State() mfaType: boolean;

  componentDidLoad() {
    logger.log('Totp Rendered');
  }

  render() {
    return <div>TOTP</div>;
  }
}
