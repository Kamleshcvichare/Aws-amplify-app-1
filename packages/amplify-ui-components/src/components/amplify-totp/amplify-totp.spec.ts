import { newSpecPage } from '@stencil/core/testing';
import { AmplifyTOTP } from './amplify-totp';

describe('amplify-totp spec:', () => {
  describe('Component logic ->', () => {
    let totp;

    beforeEach(() => {
      totp = new AmplifyTOTP();
    });

    it('should render authData to null by default', () => {
      expect(totp.authData).toEqual(null);
    });

    it('should render `MFATypes` to undefined by default', () => {
      expect(totp.MFATypes).toBeUndefined();
    });
  });
  describe('Render logic ->', () => {
    it('should render a `less than 2 mfa types available` message by default', async () => {
      const page = await newSpecPage({
        components: [AmplifyTOTP],
        html: `<amplify-totp></amplify-totp>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
