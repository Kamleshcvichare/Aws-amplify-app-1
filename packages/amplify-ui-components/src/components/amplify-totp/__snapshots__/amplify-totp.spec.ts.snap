// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-totp spec: Render logic -> should render a \`less than 2 mfa types available\` message by default 1`] = `
<amplify-totp>
  <div>
    TOTP
  </div>
</amplify-totp>
`;
